autoload/ale/references.vim

```diff
@@ -51,18 +53,44 @@ function! ale#references#HandleLSPResponse(conn_id, response) abort
 
         if type(l:result) is v:t_list
             for l:response_item in l:result
-                call add(l:item_list, {
-                \ 'filename': ale#path#FromURI(l:response_item.uri),
-                \ 'line': l:response_item.range.start.line + 1,
-                \ 'column': l:response_item.range.start.character + 1,
-                \})
+                let l:col = l:response_item.range.start.character + 1
+                let l:lnum = l:response_item.range.start.line + 1
+                let l:filename = ale#path#FromURI(l:response_item.uri)
+
+                if g:ale_use_quickfix_findref
+                    let l:cwd = getcwd()
```

Comment:
Have a look at the custom linter output which suggests an alternative to `getcwd()` that works if you change buffers before an asynchronous callback is fired.